src/hidimstat/dcrt.py

```diff
@@ -1,455 +1,491 @@
     },
 ):
     """
-    Standard Lasso Distillation for least squares regression.
+    Standard Lasso distillation for least squares regression.
 
-    This function implements the distillation process following :footcite:t:`liu2022fast`
-    section 2.3. It distills both X[:, idx] and y to compute test statistics.
-    It's based on least square loss regression.
+    This function fits a Lasso model (or other estimator) to predict X[:, idx]
+    from the remaining features, and optionally fits a model to predict y from
+    the remaining features.
+    Used as a helper for parallel feature-wise distillation in dCRT.
 
     Parameters
     ----------
-    X : {array-like, sparse matrix} of shape (n_samples, n_features)
-        The input samples.
-    y : array-like of shape (n_samples,)
-        The target values.
     idx : int
         Index of the variable to be tested.
-    coef_full : array-like of shape (n_features,), default=None
-        Pre-computed coefficients for y prediction. If None, computed via Lasso.
-    sigma_X : {array-like, sparse matrix} of shape (n_features, n_features), default=None
-        The covariance matrix of X.
-    n_jobs : int, default=1
-        Number of CPUs to use.
+    X : array-like of shape (n_samples, n_features)
+        The input data matrix.
+    y : array-like of shape (n_samples,)
+        The target values.
+    estimator : sklearn estimator
+        The estimator used for distillation and prediction.
+    sigma_X : bool, default=False
+        If True, use Lasso for X distillation; if False, skip X distillation.
     fit_y : bool, default=False
-        Whether to fit y using Lasso when coef_full is None.
-    random_state : int, default=42
-        Random seed for reproducibility.
-    params_lasso_distillation_x : dict
-        Parameters for main Lasso estimation or crossvalidation Lasso, including:
-        - alpha : float, optional - L1 regularization strength. If None, determined by CV.
-        - n_alphas : int, default=0 - Number of alphas for cross-validation.
-        - alphas : array-like, default=None - List of alpha values to try in CV.
-        - alpha_max_fraction : float, default=0.5 - Scale factor for alpha_max.
-    params_lasso_distillation_y : dict
-        Same as params_lasso_distillation_x.
+        Whether to fit y using Lasso when coefficient is None.
+    params_lasso_distillation_x : dict, optional
+        Parameters for Lasso estimation or cross-validated Lasso for X distillation.
+        (for details see the description of parameter *params_lasso_screening*
+        of DOCRT)
 
     Returns
     -------
-    X_res : ndarray of shape (n_samples,)
-        The residuals after X distillation.
-    sigma2 : float
-        The estimated variance of the residuals.
-    y_res : ndarray of shape (n_samples,)
-        The residuals after y distillation.
+    clf_x : estimator or None
+        Fitted Lasso model for X distillation, or None if not used.
```

Comment:
Why `clf_x` necessarily be a Lasso model, given that `estimator` can be any sklearn compatible estimator ?

Reply:
Then we need to update the docstrings to make them consistent + open an issue for that topic in particular.

Reply:
See issue #328, for continuing the conversation.